File: js/test/playerSpec.js
const expect = require('chai').expect;
const Player = require('../public/src/player.js')

describe('Player', function() {
  var player;

  beforeEach(function() {
    player = new Player();
  })

  describe('Initialisation', function() {
    it('starts with 100 health', function() {
      expect(player.health).equal(100);
    })

    it('starts with no weapons', function() {
      expect(player.weapons).to.deep.equal([]);
    })

    it('starts with 10 points of hunger', function() {
      expect(player.hunger).equal(10);
    })

    it('starts with 10 points of thirst', function() {
      expect(player.thirst).equal(10);
    })
  })

  describe('#surviveDay', function() {
    it('the player loses hunger points after a day', function() {
      player.surviveDay();
      expect(player.hunger).equal(9);
    })
  })
})

File: ruby/spec/player_spec.rb
require 'player'

describe Player do
  let(:player) { described_class.new }

  it 'starts with max health' do
    expect(player.health).to eq Player::MAX_HEALTH
  end

  it 'can have starting health set lower manually' do
    weak_player = Player.new('Dave', 50, 10)
    expect(weak_player.health).to eq 50
  end

  it 'reduces food levels after a new day' do
    player.survive_day
    expect(player.hunger).to eq Player::MAX_HUNGER - 1
  end
end
